code/game/objects/structures/mysterymachine.dm

```diff
@@ -0,0 +1,98 @@
+GLOBAL_LIST_EMPTY(possibleMMItems)
+
+/obj/structure/mysterymachine
+	name = "Mysterious Generation Machine"
+	desc = "Within this machine is the greatest power known to all of existence: unlimited creation! Whether built by an omnipotent god or an advanced omniscient intelligence created by an alien civilization, you do not know. What you do know however is that all you must do to receive whatever your heart desires is to type it in and- wait a second... What immature deviant replaced the keyboard with a lever?!?"
+	icon_state = "mysterymachine"
+	anchored = TRUE
+	density = TRUE
+	var/processing = FALSE
+	var/item_input = ""
+	var/inputSize = 32
+	var/itemFound = FALSE
+	var/itemText = ""
+	var/lastText = ""
+	var/firstText = ""
+
+/obj/structure/mysterymachine/Initialize()
+	. = ..()
+	resistance_flags |= INDESTRUCTIBLE
+	set_light(2, 2, "#FFFFFF")
+	if(!GLOB.possibleMMItems.len)
+		var/list/MMItems_types_list = subtypesof(/obj/item)
```

Comment:
`var/list/MMItems_types_list = subtypesof(/obj/item)`

`subtypesof(/obj/item)`


![](https://memegenerator.net/img/instances/80015999/concern.jpg)

![](https://i.redd.it/bix7bbtzp9h11.jpg)

![](http://www.dreams.metroeve.com/wp-content/uploads/2017/03/dreams.metroeve_concern-dreams-meaning.jpg)

![](https://upload.wikimedia.org/wikipedia/en/3/32/Concern_worldwide_logo.gif)

![](http://www.coloradoconcern.com/wp-content/uploads/2018/02/Colorado-Concern-Logo-500.png)

![](https://i.kym-cdn.com/photos/images/original/001/274/502/5b5.png)

![](https://i.imgur.com/y2wcXNJ.jpg)

![](https://i.imgur.com/GmlvwmI.jpg)

![](https://i.kym-cdn.com/photos/images/original/001/363/857/d01.jpg)

Reply:
this may have been marked as spam but it is educational and i highly advise you show comment

Reply:
and i know a rookie coder will look at this down the line and think "Well its safe to use here what about elsewhere"

no it is not

it is not safe

do not do this without extreme care and caution